server/hedley/modules/custom/hedley_reports/scripts/generate-nutrition-group-completion-data.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+/**
+ * @file
+ * Generates completion data for Nutrition group encounters.
+ *
+ * Execution: drush scr
+ *   profiles/hedley/modules/custom/hedley_reports/scripts/generate-nutrition-group-completion-data.php.
+ */
+
+if (!drupal_is_cli()) {
+  // Prevent execution from browser.
+  return;
+}
+
+// Get the last node id.
+$nid = drush_get_option('nid', 0);
+
+// Get the number of nodes to be processed.
+$batch = drush_get_option('batch', 50);
+
+// Flag to generate data only if it was not generated already.
+$exclude_set = drush_get_option('exclude_set', FALSE);
+
+// Get allowed memory limit.
+$memory_limit = drush_get_option('memory_limit', 500);
+
+$type = 'attendance';
+$base_query = new EntityFieldQuery();
+$base_query
+  ->entityCondition('entity_type', 'node')
+  ->entityCondition('bundle', $type)
+  ->fieldCondition('field_attended', 'value', TRUE)
+  ->propertyCondition('status', NODE_PUBLISHED);
+
+if ($exclude_set) {
+  $base_query->addTag('exclude_set_reports_data');
+}
+
+$count_query = clone $base_query;
+$count_query->propertyCondition('nid', $nid, '>');
+$count = $count_query->count()->execute();
+
+if ($count == 0) {
+  drush_print("There are no nodes of type $type in DB.");
+  exit;
+}
+
+$total = 0;
+drush_print("$count nodes of type $type located.");
+
+while (TRUE) {
+  $query = clone $base_query;
+  if ($nid) {
+    $query->propertyCondition('nid', $nid, '>');
+  }
+
+  $result = $query
+    ->range(0, $batch)
+    ->execute();
+
+  if (empty($result['node'])) {
+    // No more items left.
+    break;
+  }
+
+  $ids = array_keys($result['node']);
+  $nodes = node_load_multiple($ids);
+  foreach ($nodes as $node) {
+    $completion_data = hedley_reports_generate_completion_data_for_nutrition_group_encounter($node);
+    $node->field_reports_data[LANGUAGE_NONE][0]['value'] = json_encode($completion_data);
+    node_save($node);
```

Comment:
try-catch, error handling perhaps?

Reply:
@AronNovak 

Similar to https://github.com/TIP-Global-Health/eheza-app/pull/1283#discussion_r1730425233